src/time_series_metadata_store.jl

```diff
@@ -0,0 +1,1004 @@
+    name::String;
+    features...,
+)
+    rows = _try_time_series_metadata_by_full_params(
+        store,
+        owner,
+        time_series_type,
+        name,
+        "metadata";
+        features...,
+    )
+    len = length(rows)
+    if len == 0
+        return nothing
+    elseif len == 1
+        return _deserialize_metadata(rows[1].metadata)
+    else
+        throw(ArgumentError("Found more than one matching time series: $len"))
+    end
+end
+
+function _try_has_time_series_metadata_by_full_params(
+    store::TimeSeriesMetadataStore,
+    owner::TimeSeriesOwners,
+    time_series_type::Type{<:TimeSeriesData},
+    name::String;
+    features...,
+)
+    row = _try_time_series_metadata_by_full_params(
+        store,
+        owner,
+        time_series_type,
+        name,
+        "id";
+        features...,
+    )
+    return !isempty(row)
+end
+
+function _try_time_series_metadata_by_full_params(
+    store::TimeSeriesMetadataStore,
+    owner::TimeSeriesOwners,
+    time_series_type::Type{<:TimeSeriesData},
+    name::String,
+    column::String;
+    features...,
+)
+    where_clause = _make_where_clause(
+        owner;
+        time_series_type = time_series_type,
+        name = name,
+        require_full_feature_match = true,
+        features...,
+    )
+    query = "SELECT $column FROM $METADATA_TABLE_NAME WHERE $where_clause"
+    return Tables.rowtable(_execute(store, query))
+end
+
+function compare_values(
+    x::TimeSeriesMetadataStore,
+    y::TimeSeriesMetadataStore;
+    compare_uuids = false,
+    exclude = Set{Symbol}(),
+)
+    # Note that we can't compare missing values.
+    owner_uuid = compare_uuids ? ", owner_uuid" : ""
+    query = """
+        SELECT id, metadata, time_series_uuid $owner_uuid
+        FROM $METADATA_TABLE_NAME ORDER BY id
+    """
+    table_x = Tables.rowtable(_execute(x, query))
+    table_y = Tables.rowtable(_execute(y, query))
+    return table_x == table_y
+end
+
+### Non-TimeSeriesMetadataStore functions ###
+
+_convert_ts_type_to_string(ts_type::Type{<:TimeSeriesData}) = string(nameof(ts_type))
+
+function _deserialize_metadata(text::String)
+    val = JSON3.read(text, Dict)
+    return deserialize(get_type_from_serialization_data(val), val)
+end
+
+_get_owner_category(
+    ::Union{InfrastructureSystemsComponent, Type{<:InfrastructureSystemsComponent}},
+) = "Component"
+_get_owner_category(::Union{SupplementalAttribute, Type{<:SupplementalAttribute}}) =
+    "SupplementalAttribute"
+_get_time_series_category(::Type{<:Forecast}) = "Forecast"
+_get_time_series_category(::Type{<:StaticTimeSeries}) = "StaticTimeSeries"
+
+function _make_feature_filter(; features...)
+    data = _make_sorted_feature_array(; features...)
+    return join((["metadata->>'\$.features.$k' = '$v'" for (k, v) in data]), "AND ")
+end
+
+function _make_features_string(features::Dict{String, <:Any})
+    key_names = sort!(collect(keys(features)))
+    data = [Dict(k => features[k]) for k in key_names]
```

Comment:
For the record, I had questions about whether constructing one-element dictionaries was inefficient compared to lists, and the answer is no:
```julia
julia> @btime JSON3.write([Dict("a" => 1), Dict("b" => "c")])
  2.185 μs (26 allocations: 1.58 KiB)
"[{\"a\":1},{\"b\":\"c\"}]"

julia> @btime JSON3.read("[{\"a\":1},{\"b\":\"c\"}]")
  2.079 μs (10 allocations: 832 bytes)
2-element JSON3.Array{JSON3.Object, Base.CodeUnits{UInt8, String}, Vector{UInt64}}:
 {
   "a": 1
}
 {
   "b": "c"
}

julia> @btime JSON3.write([["a", 1], ["b", "c"]])
  4.554 μs (39 allocations: 1.41 KiB)
"[[\"a\",1],[\"b\",\"c\"]]"

julia> @btime JSON3.read("[[\"a\",1],[\"b\",\"c\"]]")
  2.268 μs (10 allocations: 832 bytes)
2-element JSON3.Array{JSON3.Array, Base.CodeUnits{UInt8, String}, Vector{UInt64}}:
 Any["a", 1]
 ["b", "c"]
```